GitCommands/Submodules/SubmoduleStatusProvider.cs

```diff
@@ -100,7 +100,7 @@ public void UpdateSubmodulesStatus(bool updateStatus, string workingDirectory, s
                 // First task: Gather list of submodules on a background thread.
 
                 // Don't access Module directly because it's not thread-safe.  Use a thread-local version:
-                var threadModule = new GitModule(workingDirectory);
+                var threadModule = gitModule;
```

Comment:
I think that with this change the whole Lazy changes are unnecessary. Aren't they?

I don't understand how the Lazy encoding would help, for it is always evaluated when passed through and it is already lazily created.

Reply:
This commit (including fixup) avoids recreating the latest GitModule only, so only one git call.

The encoding is only evaluated when there are changes to submodules. The normal scenario is that there are changes to few submodules only (if there are no submodule changes, the submodule status is not evaluated at all (just structure)).
The encoding is only needed when there is a change for the submodule.

Reply:
Thanks, I see it now. Perhaps FilesEncoding don't have to be declared as Lazy, it is enough to pass it as Lazy or Func<Encoding> where needed.

Reply:
Func would work too, I declared it as Lazy as I found it simpler.  (Lazy is hardly usedin GE right now so it is basically a new "pattern").
Do request to change it?

Reply:
I mean:

```csharp
public Encoding FilesEncoding => EffectiveConfigFile.FilesEncoding ?? new UTF8Encoding(false);

public static IEnumerable<Patch> 
CreatePatchesFromString([NotNull] string patchText, [NotNull] Func<Encoding> filesContentEncoding)
        

var patches = PatchProcessor.CreatePatchesFromString(output, () => encoding ?? FilesEncoding).ToList();
```
or
```csharp
public Encoding FilesEncoding => EffectiveConfigFile.FilesEncoding ?? new UTF8Encoding(false);

public static IEnumerable<Patch> 
CreatePatchesFromString([NotNull] string patchText, [NotNull] Lazy<Encoding> filesContentEncoding)
        

var patches = PatchProcessor.CreatePatchesFromString(output, new Lazy(() => encoding ?? FilesEncoding)).ToList();


```

Reply:
I prefer to use Lazy. Will change to your proposal, it further emphasis why the change is done. If most "expensive" fields were Lazy, it would be a different setup.

Reply:
Changed - fixup